src/lib/LUA/tx_devLUA.cpp

```diff
@@ -578,7 +588,12 @@ static void registerLuaParameters()
       uint8_t newSwitchMode = adjustSwitchModeForAirRate(
         (OtaSwitchMode_e)config.GetSwitchMode(), get_elrs_airRateConfig(actualRate)->PayloadLength);
       // If the switch mode is going to change, block the change while connected
-      if (newSwitchMode == OtaSwitchModeCurrent || connectionState == disconnected)
+      bool isDisconnected = connectionState == disconnected;
+      // Don't allow the switch mode to change if the TX is in mavlink mode
+      // Wide switchmode is not compatible with mavlink, and the switchmode is
+      // auto configuredwhen entering mavlink mode
+      bool isMavlinkMode = config.GetLinkMode() == TX_MAVLINK_MODE;
+      if (newSwitchMode == OtaSwitchModeCurrent || (isDisconnected && !isMavlinkMode))
```

Comment:
Should we look at not populating Switch and TLM options from Lua when in Mav mode?

Reply:
FYI for anyone following... Im doing this as a side quest and will not hold up this PR.